### ISSUE TITLE
App won't run on ios simulator/ resource fork, Finder information, or similar detritus not allowed

### ISSUE THREAD

**Issue by emmatuovinen** [flutter/flutter#49290]
Hi! I'm having trouble running the default flutter app on the ios simulator. I managed to run the 'first flutter app' specified on the flutter web page (https://flutter.dev/docs/get-started/codelab) successfully earlier today and now (without making any changes into it) it won't run anymore either. This is the error message I receive: 

Runner.app: resource fork, Finder information, or similar detritus not
    allowed
    Command CodeSign failed with a nonzero exit code
    note: Using new build system
    note: Planning build
    note: Constructing build description
    warning: Capabilities for Signing & Capabilities may not function correctly
    because its entitlements use a placeholder team ID. To resolve this, select
    a development team in the Runner editor. (in target 'Runner' from project
    'Runner')

Could not build the application for the simulator.
Error launching application on iPhone 11.

Am I missing something here? Thanks!

**Comment by VladyslavBondarenko** [comment]
Hi @emmatuovinen 
Related to https://github.com/flutter/flutter/issues/28077
Does the last comment help you?
Thank you

**Comment by jmagman** [comment]
I can't tell the path of the Runner.app that failed to sign (did you leave it off the first line of the error message)?

What's the output of:
```
$ xattr -lr <path/to/Runner.app>
```

**Comment by emmatuovinen** [comment]
@VladyslavBondarenko your solution seems to work at least with with my initial attempts with 2 different projects, thanks! It looks like I need to run the xattr -cr app_name command every time the simulator has been disconnected and is restarted?
@jmagman I did leave the first line out, the complete path was /Users/emma/Documents/flutter_app/build/ios/Debug-iphonesimulator/Runner.app

**Comment by jmagman** [comment]
@emmatuovinen  What's the output of:
```
$ xattr -lr /Users/emma/Documents/flutter_app/build/ios/Debug-iphonesimulator/Runner.app
```

**Comment by emmatuovinen** [comment]
@jmagman  I changed the path but the updated version's output is this: 
/Users/emma/Documents/Coding/flutterApps/derp_app/build/ios/Debug-iphonesimulator/Runner.app: com.apple.FinderInfo:
00000000  00 00 00 00 00 00 00 00 20 10 00 00 00 00 00 00  |........ .......|
00000010  00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  |................|
00000020

**Comment by PolymathWhiz** [comment]
This fixed my issue - https://developer.apple.com/library/archive/qa/qa1940/_index.html

**Comment by jmagman** [comment]
I have no idea what would be adding a com.apple.FinderInfo xattr to Runner.app.  This sounds like an environment issue.

The tool could query the xattrs and then `xattr -d` delete them at some point in the installation process.

**Comment by jmagman** [comment]
I'd like to fix this one but I don't think we should be getting rid of xattrs that Finder is adding, it could be adding them because the build directory is on an external drive, or a network drive, or adding some mime type metadata or sync info or really anything undocumented.

**Comment by jmagman** [comment]
Changed my mind, it's annoying and removing the xattr is probably harmless.  Speculatively fixing with https://github.com/flutter/flutter/issues/26123 since I don't know exactly what files have the attributes before code signing.

**Comment by rwebster2012** [comment]
> This fixed my issue - https://developer.apple.com/library/archive/qa/qa1940/_index.html



> This also fixed my issue. 

Just quoting what what essentially said in the link. Just run these two lines in your terminal:

$ xattr -lr <path_to_app_bundle>
$ xattr -cr <path_to_app_bundle>

**Comment by jmagman** [comment]
@rwebster2012 If you (or anyone) sees this again, can you try the [master channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#how-to-change-channels) to test out https://github.com/flutter/flutter/pull/54488?  If that fix doesn't work then this issue needs to be reopened.

**Comment by nicovak** [comment]
@jmagman LGTM. I Just tried master and the error is gone. Switching back to stable and still have the issue.

**Comment by jmagman** [comment]
@nicovak Thanks for checking! 👍

**Comment by pintom** [comment]
@jmagman I can also confirm that the error is gone on the master channel. Thank you

**Comment by antonioJASR** [comment]
For anyone using the stable channel. I fix it with the following instructions:

- Open Xcode and Select the "Runner" icon in Project Navigator

<img width="476" alt="Screen Shot 2020-05-01 at 18 17 04" src="https://user-images.githubusercontent.com/1143701/80848181-01f12880-8bd8-11ea-88f5-bb76728f7f01.png">

- Go to **Build Phases** -> **Run Script** and add `xattr -cr .` after this line:
`/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build`

<img width="1063" alt="File Edit" src="https://user-images.githubusercontent.com/1143701/80848225-2ea54000-8bd8-11ea-99d5-44db39fdcf3a.png">

That's it

**Comment by owenselles** [comment]
@antonioJASR that fixed my issue thanks!

**Comment by jmagman** [comment]
https://github.com/flutter/flutter/pull/54488 is on [1.18.0-6.0.pre](https://flutter.dev/docs/development/tools/sdk/releases?tab=macos) (so currently the master, dev, and beta channels).  If anyone is seeing this issue on a version later than that, please let us know so we can investigate further.

**Comment by Arthur-Rankin** [comment]
I received this error today with exact sam reproduction and I was able to fix with with:

**Comment by a7medpa** [comment]
> For anyone using the stable channel. I fix it with the following instructions:
> 
> * Open Xcode and Select the "Runner" icon in Project Navigator
> 
> <img alt="Screen Shot 2020-05-01 at 18 17 04" width="476" src="https://user-images.githubusercontent.com/1143701/80848181-01f12880-8bd8-11ea-88f5-bb76728f7f01.png">
> 
> * Go to **Build Phases** -> **Run Script** and add `xattr -cr .` after this line:
>   `/bin/sh "$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build`
> 
> <img alt="File Edit" width="1063" src="https://user-images.githubusercontent.com/1143701/80848225-2ea54000-8bd8-11ea-99d5-44db39fdcf3a.png">
> 
> That's it

I tried this and I get this as a result:
xattr: [Errno 13] Permission denied: './Pods/FirebaseFirestore/Firestore/core/src/remote/grpc_root_certificates_generated.cc'

I tried all the other solutions provided as well such as 
`$ xattr -lr <path_to_app_bundle>`
`$ xattr -cr <path_to_app_bundle>`
Which just gives me the standard error:
/build/ios/Debug-iphonesimulator/Runner.app: resource fork, Finder information, or similar detritus not allowed

😢

**Comment by jmagman** [comment]
The build phase script in Xcode runs as your user (an Xcode policy, not a Flutter one). If your user doesn't have permission to edit the directory your Flutter project all kinds of things aren't going to work.  `flutter create` complains if you run with `sudo`.

**Comment by DomingoMG** [comment]
It worked for me with `flutter clean`

**Comment by ziz19** [comment]
Every time I make a build, I need to execute either `xattr -cr .` or `flutter clean`. Is there a way to permanently eliminate the security issue?

**Comment by jmagman** [comment]
@ziz19 This is the first I've heard it's not working past https://github.com/flutter/flutter/pull/54488 (available on 1.18 and later), you can see above confirming the fix worked for them.
Can you file a new issue and completely fill out the template?  We want to take a look.

**Comment by ziz19** [comment]
> @ziz19 This is the first I've heard it's not working past #54488 (available on 1.18 and later), you can see above users confirming the fix worked for them (other than unrelated file permissions issues).
> Can you file a new issue and completely fill out the template? We want to take a look.

sorry that I didn't notice this until today. I have filed a new issue #79689
